src/_pytest/python.py

```diff
@@ -571,20 +587,20 @@ def setup(self):
             func = partial(_call_with_optional_argument, teardown_module, self.obj)
             self.addfinalizer(func)
 
-    def _recurse(self, dirpath):
+    def _recurse(self, dirpath: py.path.local) -> bool:
         if dirpath.basename == "__pycache__":
             return False
         ihook = self.gethookproxy(dirpath.dirpath())
         if ihook.pytest_ignore_collect(path=dirpath, config=self.config):
-            return
+            return False
```

Comment:
Had to make it a proper False - shouldn't matter to `py.path.local` recurse as long as it's falsey.

Reply:
Also conflicts with https://github.com/pytest-dev/pytest/pull/6461, where I've aligned that already.